### ISSUE TITLE
Airplay 2 - paired receivers support - stereo

### ISSUE THREAD

**Issue by pawelka** [ejurgensen/forked-daapd#704]
Observations: 
- Paired receivers are discovered separately. 
- When playing audio only one receives. 
- When trying to select second receiver in parallel first one turn off with log:
>[2019-02-23 21:17:32] [  LOG]     raop: Device 'HomePod Left' closed RTSP connection
[2019-02-23 21:17:32] [DEBUG]     raop: Cleaning up failed session (deferred) on device 'HomePod Left'
[2019-02-23 21:17:32] [DEBUG]   player: Callback from AirPlay to device_streaming_cb
[2019-02-23 21:17:32] [  LOG]   player: The AirPlay device 'HomePod Left' FAILED

- After unpairing receivers, I'm able to stream to two receivers.
- On the phone paired receivers are discovered once under one name.

I'm not sure if it's bug or new feature.

**Comment by ejurgensen** [comment]
Thanks for reporting this. I don't think there is much to do about it - or at least I wouldn't know what to do, given that the internals of Airplay 2 aren't known. I will keep the issue open for a while, maybe someone turns up with an idea.

**Comment by pawelka** [comment]
More info from avahi. 

When speakers are paired:
> = enp0s4 IPv4 HomePod Right                                 _airplay._tcp        local
   hostname = [HomePod-Right.local]
   address = [192.168.2.102]
   port = [7000]
   txt = ["vv=2" "osvers=12.1.3" "srcvers=375.3" "pk=a10549364147630c7ee7dda909013408621c7876979cf280e8c647b8252722bc" "psi=BC363713-0C96-4833-8B0B-873A8C405579" "pi=8f3afcdb-14c4-4d50-861c-8499f2d418e6" "protovers=1.1" "model=AudioAccessory1,1" "tsid=6BA31C4B-5D14-57DC-9B4A-8C919D3E4068" "gpn=Living Room" "gcgl=1" "igl=1" "gid=6BA31C4B-5D14-57DC-9B4A-8C919D3E4068" "flags=0x1a404" "features=0x4A7FCA00,0x156BD0" "deviceid=D4:90:9C:C7:80:6E" "acl=0"]
= enp0s4 IPv4 HomePod Left                                  _airplay._tcp        local
   hostname = [HomePod-Left.local]
   address = [192.168.2.101]
   port = [7000]
   txt = ["vv=2" "osvers=12.1.3" "srcvers=375.3" "pk=598299e0cefa217a565db72b4fa402e352eb1d2248f51d121ba25c001f2daf8f" "psi=B59CEEE6-DD2B-41ED-A99C-085D8FF24968" "pi=952e77d0-f09f-47b0-955e-2ffdc4f2fb77" "protovers=1.1" "model=AudioAccessory1,1" "tsid=6BA31C4B-5D14-57DC-9B4A-8C919D3E4068" "gpn=Living Room" "gcgl=1" "igl=1" "gid=6BA31C4B-5D14-57DC-9B4A-8C919D3E4068" "flags=0x18404" "features=0x4A7FCA00,0x156BD0" "deviceid=D0:81:7A:E3:A8:AF" "acl=0"]

When speakers are unpaired:
>    = enp0s4 IPv4 HomePod Left                                  _airplay._tcp        local
      hostname = [HomePod-Left.local]
      address = [192.168.2.101]
      port = [7000]
      txt = ["vv=2" "osvers=12.1.3" "srcvers=375.3" "pk=598299e0cefa217a565db72b4fa402e352eb1d2248f51d121ba25c001f2daf8f" "psi=B59CEEE6-DD2B-41ED-A99C-085D8FF24968" "pi=952e77d0-f09f-47b0-955e-2ffdc4f2fb77" "protovers=1.1" "model=AudioAccessory1,1" "gcgl=1" "igl=1" "gid=6B15CAD3-4593-4499-ABB7-B367A927E006" "flags=0x18404" "features=0x4A7FCA00,0x156BD0" "deviceid=D0:81:7A:E3:A8:AF" "acl=0"]
   = enp0s4 IPv4 HomePod Right                                 _airplay._tcp        local
      hostname = [HomePod-Right.local]
      address = [192.168.2.102]
      port = [7000]
      txt = ["vv=2" "osvers=12.1.3" "srcvers=375.3" "pk=a10549364147630c7ee7dda909013408621c7876979cf280e8c647b8252722bc" "psi=BC363713-0C96-4833-8B0B-873A8C405579" "pi=8f3afcdb-14c4-4d50-861c-8499f2d418e6" "protovers=1.1" "model=AudioAccessory1,1" "gcgl=1" "igl=1" "gid=E515D378-262E-4850-B73C-E864CCD4B8C9" "flags=0x18404" "features=0x4A7FCA00,0x156BD0" "deviceid=D4:90:9C:C7:80:6E" "acl=0"]
> 
The difference is:
> "tsid=6BA31C4B-5D14-57DC-9B4A-8C919D3E4068" "gpn=Living Room" both shares the same value 

I'm not sure how exactly airplay works, but my guess is that after pairing devices should share session. 

Is it possible to have some blind fix with session searching? Means if tsid is the same same session should be shared.

**Comment by ejurgensen** [comment]
I'm not sure what that fix would be? Airplay 1 does not have a session concept, it works point-to-point. So all forked-daapd can do is open a connection to one or more devices. It can connect to both devices if it sees that they have the same tsid, but according to your first post it will just mean that one of the devices starts - the other one will turn off?

**Comment by pawelka** [comment]
Yes. 

>[2019-02-23 21:17:32] [ LOG] raop: Device 'HomePod Left' closed RTSP connection
[2019-02-23 21:17:32] [DEBUG] raop: Cleaning up failed session (deferred) on device 'HomePod Left'

After reading second line I guess that there is some session.

**Comment by pawelka** [comment]
I check in wireshark that during playing packages (stream) goes to both speakers.
